examples/linear_model/plot_ridge_coeffs.py

```diff
@@ -1,89 +1,272 @@
+import numpy as np
+
 from sklearn.linear_model import Ridge
 from sklearn.metrics import mean_squared_error
 
 clf = Ridge()
 
+# Generate values for `alpha` that are evenly distributed on a logarithmic scale
+alphas = np.logspace(-3, 4, 200)
+coefs = []
+errors_coefs = []
+
+# Train the model with different regularisation strengths
+for a in alphas:
+    clf.set_params(alpha=a).fit(X, y)
+    coefs.append(clf.coef_)
+    errors_coefs.append(mean_squared_error(clf.coef_, w))
+
+# %%
+# Note: In a real setting you should use :class:`~sklearn.linear_model.RidgeCV`
+# for training a Ridge regression with built-in cross-validation, but for the
+# purpose of demonstration we will keep it simple here.
+
+# %%
+# Plotting trained Coefficients and Mean Squared Errors
+#######################################################
+# We now plot the 10 different regularized coefficients as a function of the
+# regularization parameter `alpha` where each color represents a different
+# coefficient.
+#
+# On the right-hand-side, we plot how the errors of the coefficients from the
+# estimator change as a function of regularization.
+import matplotlib.pyplot as plt
+import pandas as pd
+
+alphas = pd.Index(alphas, name="alpha")
+coefs = pd.DataFrame(coefs, index=alphas, columns=[f"Feature {i}" for i in range(10)])
+errors = pd.Series(errors_coefs, index=alphas, name="Mean squared error")
+
+fig, axs = plt.subplots(1, 2, figsize=(20, 6))
+
+coefs.plot(
+    ax=axs[0],
+    logx=True,
+    ylabel="Ridge coefficient values",
+    title="Ridge coefficients as a function of the regularization strength",
+)
+_ = errors.plot(
+    ax=axs[1],
+    logx=True,
+    ylabel="Mean squared error",
+    title="Coefficient error as a function of the regularization strength",
+)
+
+# Interpreting the plots
+########################
+# The plot on the left-hand side shows how the regularization strength (`alpha`)
+# affects the Ridge regression coefficients. Smaller values of `alpha` (weak
+# regularization), allow the coefficients to closely resemble the true
+# coefficients (`w`) used to generate the data set. This is because no additional
+# noise was added to our artificial data set. As `alpha` increases, the
+# coefficients shrink towards zero, gradually reducing the impact of the
+# features that were formerly more significant.
+#
+# The right-hand side plot shows the `mean squared error` between the
+# coefficients found by the model and the true coefficients (`w`). It provides a
+# measure that relates to how exact our ridge model is in comparison to the true
+# generative model. A low error means that it found coefficients closer to the
+# ones of the true generative model. In this case, since our toy data set was
+# non-noisy, we can see that the least regularized model retrieves coefficients
+# closest to the true coefficients (`w`) (error is close to 0).
+
+# %%
+# Ridge Regularization on noisy data set
```

Comment:
Let's remove this section and onwards. I computed the model score and you get a R2 of 0.5 for non-penalized model and then it always decreases when alpha increases. It means that it is not correlated with the value of beta so the story does not hold here.

I don't want to block the refactoring of the example since the first part is already a net improvement.
We only need to keep in mind that we should revisit this example to add a section on a realistic dataset where we want to show that tuning the hyperparameter `alpha` is mandatory with `Ridge`.

Reply:
Alright, I only modified a part of the conclusion to be included here.